net/tls.go

```diff
@@ -0,0 +1,19 @@
+package net
+
+import "crypto/tls"
+
+// This is from gtank's cryptopasta defaults
+// https://github.com/gtank/cryptopasta
+func DefaultTLSConfig() *tls.Config {
+	return &tls.Config{
+		// Avoids most of the memorably-named TLS attacks
+		MinVersion: tls.VersionTLS12,
+		// Causes servers to use Go's default ciphersuite preferences,
+		// which are tuned to avoid attacks. Does nothing on clients.
+		PreferServerCipherSuites: true,
+		// Only use curves which have constant-time implementations
+		CurvePreferences: []tls.CurveID{
+			tls.CurveP256,
```

Comment:
I would suggest adding `tls.X25519` here, before `tls.CurveP256` (this was specifically recommended by @gtank despite what's on cryptopasta. Also Curve25519 rocks)